test/types/sync/ferguson/sync-record-no-default-init.chpl

```diff
@@ -0,0 +1,136 @@
+class C { }
+record R {
+  var x: int = 0;
+  var ptr: shared C = new shared C();
+  // intentionally no init() -- no default init
```

Comment:
Might be clearer to say "no argument-less init" or something, but I won't lose sleep if you don't